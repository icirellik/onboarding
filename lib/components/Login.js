import React from 'react';
import { Link } from 'react-router-dom';

const Login = () => {
  return (
    <div>
      <Link className="btn btn-primary" to="/pin-entry">Manager</Link>
      <Link className="btn btn-primary" to="/pin-entry">Employee</Link>
    </div>
  );
};

export default Login;
