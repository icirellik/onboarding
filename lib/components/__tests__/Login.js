import React from 'react';
import Login from '../Login';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Login', () => {

  it('renders correctly', () => {

    const wrapper = shallow(
      <Login />
    );

    console.log(wrapper.debug())

    expect(wrapper.find('Link').length).toBe(2);
    expect(wrapper).toMatchSnapshot();

  });

});
