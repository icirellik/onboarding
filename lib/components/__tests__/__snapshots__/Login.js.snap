// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          className="btn btn-primary"
          replace={false}
          to="/pin-entry"
        >
          Manager
        </Link>,
        <Link
          className="btn btn-primary"
          replace={false}
          to="/pin-entry"
        >
          Employee
        </Link>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Manager",
          "className": "btn btn-primary",
          "replace": false,
          "to": "/pin-entry",
        },
        "ref": null,
        "rendered": "Manager",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Employee",
          "className": "btn btn-primary",
          "replace": false,
          "to": "/pin-entry",
        },
        "ref": null,
        "rendered": "Employee",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            className="btn btn-primary"
            replace={false}
            to="/pin-entry"
          >
            Manager
          </Link>,
          <Link
            className="btn btn-primary"
            replace={false}
            to="/pin-entry"
          >
            Employee
          </Link>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Manager",
            "className": "btn btn-primary",
            "replace": false,
            "to": "/pin-entry",
          },
          "ref": null,
          "rendered": "Manager",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Employee",
            "className": "btn btn-primary",
            "replace": false,
            "to": "/pin-entry",
          },
          "ref": null,
          "rendered": "Employee",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
